lib/windows/app/actions/deviceActions.js

```diff
@@ -1,291 +0,0 @@
-            // user will most likely not have permission to open many of
-            // the connected usb devices.
-            resolve(deviceInfo);
-        }
-    });
-}
-
-/**
- * Get an array of available usb device info objects. See
- * `getUsbDeviceInfo` for the list of properties that the objects
- * may have.
- *
- * @returns {Promise} Promise that resolves with usb device info objects.
- */
-function getUsbDevices() {
-    return new Promise((resolve, reject) => {
-        try {
-            const usbDevices = usb.getDeviceList();
-            const promises = usbDevices.map(usbDevice => getUsbDeviceInfo(usbDevice));
-            resolve(Promise.all(promises));
-        } catch (error) {
-            reject(new Error(`Unable to get USB devices: ${error.message}`));
-        }
-    });
-}
-
-/**
- * Load devices and dispatch 'DEVICES_LOAD_SUCCESS' when devices
- * have been loaded, or 'DEVICES_LOAD_ERROR' if the loading failed.
- *
- * @returns {function(*)} Function that can be passed to redux dispatch.
- */
-export function loadDevices() {
-    return dispatch => {
-        dispatch(loadDevicesAction());
-        Promise.all([
-            getUsbDevices(),
-            getSerialPorts(),
-        ]).then(([usbDevices, serialPorts]) => {
-            dispatch(loadDevicesSuccessAction(usbDevices, serialPorts));
-        })
-        .catch(error => (
-            dispatch(loadDevicesErrorAction(`Unable to load devices: ${error.message}`))
-        ));
-    };
-}
-
-/**
- * Toggle the expanded state of the device selector. If the selector
- * is transitioning into expanded state, then the list of available
- * devices are queried in order to populate the device list.
- *
- * @returns {function(*, *)} Function that can be passed to redux dispatch.
- */
-export function toggleSelectorExpanded() {
-    return (dispatch, getState) => {
-        const state = getState();
-        if (!state.core.device.isSelectorExpanded) {
-            dispatch(loadDevices());
-        }
-        dispatch(selectorToggleExpandedAction());
-    };
-}
-
-/**
- * Select the given device by dispatching 'DEVICE_SELECTED'.
- *
- * If the selected device is a serial port, then we verify that the device
- * can be opened before dispatching 'DEVICE_SELECTED'. This is to work around
- * an issue with SEGGER J-Link devices, which will hang if we attempt to
- * connect to them when the device is not available.
- *
- * @param {Object} device The selected device.
- * @param {boolean} verifySerialPortAvailable If true, and the device has a
- *      comName, then try to open/close the serial port before dispatching
- *      'DEVICE_SELECTED'.
- * @returns {function(*)} Function that can be passed to redux dispatch.
- */
-export function selectDevice(device, verifySerialPortAvailable = true) {
-    return dispatch => {
-        if (verifySerialPortAvailable && device.comName) {
-            isPortAvailable(device.comName)
```

Comment:
When leaving a J-Link device connected to a Windows PC overnight, it will become inactive and has to be power cycled. Trying to connect to it with J-Link when it is in this state will cause a hang.

Now that this check is removed from pc-nrfconnect-core, we must remember to add this check somewhere else. Perhaps pc-nrfjprog-js could do this check.

Reply:
On second thought, pc-nrfjprog-js cannot do this check. It only deals with J-Link serial numbers, so a COM port is not available to be checked there. I suppose it has to be done by each app then.

Reply:
Yeah, I'd leave sanity checks specific to jlink to either the application or the jlink backend of the device lister. I don't think the device selector is the place to care for this.